src/routes/mod.rs

```diff
@@ -26,7 +26,12 @@ async fn embeddings(
     data: Json<Repository>,
     db: web::Data<Arc<QdrantDB>>,
     model: web::Data<Arc<Onnx>>,
-) -> impl Responder {
+) -> Result<impl Responder> {
+    let license_info = fetch_license_info(&data).await.map_err(ErrorBadRequest)?;
+    if !license_info.permissible {
+        return Err(ErrorForbidden(license_info.error.unwrap_or_default()));
```

Comment:
Hmmm I'm not sure [`403 Forbidden` is quite right](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403):

> The HTTP 403 Forbidden response status code indicates that the server understands the request but refuses to authorize it.

`400 Bad Request` makes abit more sense to me since it usually indicates that there's something malformed with the clients request. A.k.a., the client, in error, provided a repository that cannot be indexed due to its license.

This is more of a nit: happy to hear your thoughts or merge and fix latter when we have a more solid understanding of how this will be used.

Reply:
I too thought of returning a`400 BAD_REQUEST`. But then, since there isn't anything wrong the user's request, its that the server forbids processing it for some reason, I ended up returning a `403 FORBIDDEN`.